src/main/java/com/amazon/opendistroforelasticsearch/sql/request/SqlRequest.java

```diff
@@ -29,17 +29,29 @@
 
 import java.io.IOException;
 import java.util.Collections;
+import java.util.UUID;
 
 public class SqlRequest {
 
+    public static final SqlRequest NULL = new SqlRequest("Unassigned", "", null);
+
+    /** Unique request ID for tracking */
+    private final String id;
```

Comment:
I am thinking if it would be more useful to keep the id attached to the RestRequest context, not sqlRequest. That would make sure we have end-to-end (from the point of entering the plugin) tracking of requests. This is just a food for thought, no immediate action required.

Reply:
Sure. Not sure how to do this for now. Maybe we can incorporate it into your log4j proposal too. Resolving the conversation..